app/main.py

```diff
@@ -2,0 +1,109 @@
+from abc import ABC, abstractmethod
+
+
+class Player(ABC):
+    def __init__(self, nickname: str) -> None:
+        self.nickname = nickname
+
+    @abstractmethod
+    def get_rating(self) -> None:
+        pass
+
+    def player_info(self) -> None:
+        pass
+
+
+class Elf(Player):
+    def __init__(self, nickname: str, musical_instrument: str) -> None:
+        super().__init__(nickname)
+        self.musical_instrument = musical_instrument
+
+    def play_elf_song(self) -> None:
+        print(f"{self.nickname}"
+              f" is playing a song on the"
+              f" {self.musical_instrument}")
+
+
+class Dwarf(Player):
+    def __init__(self, nickname: str, favourite_dish: str) -> None:
+        super().__init__(nickname)
+        self.favourite_dish = favourite_dish
+
+    def eat_favourite_dish(self) -> None:
+        print(f"{self.nickname} is eating {self.favourite_dish}")
+
+
+class ElfRanger(Elf):
+    def __init__(self, nickname: str,
+                 musical_instrument: str,
+                 bow_level: int) -> None:
+        super().__init__(nickname, musical_instrument)
+        self.bow_level = bow_level
+
+    def get_rating(self) -> int:
+        return 3 * self.bow_level
+
+    def player_info(self) -> str:
+        return (f"Elf ranger {self.nickname}."
+                f" {self.nickname} has bow of the"
+                f" {self.bow_level} level")
+
+
+class Druid(Elf):
+    def __init__(self, nickname: str,
+                 musical_instrument: str,
+                 favourite_spell: str) -> None:
+        super().__init__(nickname, musical_instrument)
+        self._favourite_spell = favourite_spell
+
+    def get_rating(self) -> int:
+        return len(self._favourite_spell)
+
+    def player_info(self) -> str:
+        return (f"Druid {self.nickname}."
+                f" {self.nickname} has a favourite spell:"
+                f" {self._favourite_spell}")
+
+
+class DwarfWarrior(Dwarf):
+    def __init__(self, nickname: str,
+                 favourite_dish: str,
+                 hummer_level: int) -> None:
+        super().__init__(nickname, favourite_dish)
+        self.hummer_level = hummer_level
+
+    def get_rating(self) -> int:
+        return 4 + self.hummer_level
+
+    def player_info(self) -> str:
+        return (f"Dwarf warrior {self.nickname}. "
+                f"{self.nickname} has a hummer of the"
```

Comment:
There is a typo in the string 'hummer'. It should likely be 'hammer' to reflect the correct spelling.